src/components/TodosFilter/TodoFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  currentFilter: string,
+  onFilterChange?: (filter: string) => void
+};
+
+export const TodosFilter: React.FC<Props> = ({
+  currentFilter,
+  onFilterChange = () => { },
+}) => {
+  return (
+    <ul className="filters" data-cy="todosFilter">
+      <li>
+        <a
+          href="#/"
+          className={classNames({
+            selected: currentFilter === Filter.ALL,
+          })}
+          onClick={() => {
+            onFilterChange(Filter.ALL);
+          }}
```

Comment:
suggest to make one handler factory and use for each filter button.

```
const handleFilter = (filter: Filter) => () => onFilterChange(filter);
```
```suggestion
          onClick={handleFilter(Filter.ALL)}
```